src/functionaltests/java/com/ericsson/ei/utils/TestConfigs.java

```diff
@@ -75,33 +66,12 @@ public void setAuthorization() {
         System.setProperty("ldap.password", password);
         System.setProperty("ldap.user.filter", "uid={0}");
     }
-    
-    public void tearDown() {
-        if (amqpBroker != null) {
-            amqpBroker.stopBroker();
-        }
-        try {
-            conn.close();
-        } catch (Exception e) {
-            // We try to close the connection but if
-            // the connection is closed we just receive the
-            // exception and go on
-        }
 
-        if (mongoClient != null)
-            mongoClient.close();
-        if (testsFactory != null)
-            testsFactory.shutdown();
+    public static AMQPBrokerManager getBrokerFromPool(int port) {
+        return amqpBrokerPool.get(port);
```

Comment:
How should this pool work? One test class execution per isolated MessageBus or several tests can share same messagebus.
If you want one to have one isolated messagebus per test, then some sort of locking mechanism need to be implemented around this, I guess.

Reply:
It is not the concern of the test. The initializer fixes it for you. The application just need to use the port from the system property.

Reply:
Then why having AMQP broker in a hasmap when its initialized from the test and shutdown from the test? There is no meaning of having this in hash map when it only used by one test. Normally a pool of services works according:
1. Test Frameworks start and starts 20(defined by setting) services and add these services to the pool(pool has locking mechanism for every services).
2. 25 parallel tests starts and 20 of these tests occupies and hold locks for each service in pool. 5 remaining tests acquire lock on service from pool, but pool has no free service and these test is put in wait queue for acquiring lock on a service.
3. When a test is finished, test release the lock for the service and one test from the lock queue can now acquire lock on the newly lock released service.
4. And it continues in the same way for all tests that is the lock queue.
This my view of how a pool normally works.
I guess the idea was to reduce the test execution time by reducing start/boot up time of message bus, and reusing already running message buses.